File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int size = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        //if 또는 switch 사용
        for (int i = 0; i < size; i++) {
            String command = br.readLine();
            switch (command) {
                case "pop": {
                    if (stack.isEmpty()) bw.write("-1\n");
                    else bw.write(stack.pop() + "\n");
                    break;
                }
                case "size": {
                    bw.write(stack.size() + "\n");
                    break;
                }
                case "empty": {
                    if (stack.isEmpty()) bw.write("1\n");
                    else bw.write("0\n");
                    break;
                }
                case "top": {
                    if (stack.isEmpty()) bw.write("-1\n");
                    else bw.write(stack.peek() + "\n");
                    break;
                }
                default: {
                    String[] arr = command.split(" ");
                    int n = Integer.parseInt(arr[1]);
                    stack.push(n);
                }
            }
        }
        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Silver/10828. 스택/스택.kt
package baekjoon

/*
백준 10826번
 */

class Stack {
    var stack = mutableListOf<Int>()

    fun push(number: Int) {
        stack.add(number)
    }

    fun top(): Int {
        if (stack.isEmpty())
            return -1
        return stack.last()
    }

    fun size(): Int {
        return stack.size
    }

    fun empty(): Int {
        if (stack.isEmpty())
            return 1
        return 0
    }

    fun pop(): Int {
        if (stack.isEmpty())
            return -1
        return stack.removeLast()
    }
}

fun main(args: Array<String>) {
    val size = readLine()!!.toInt()
    val stack = Stack()
    for (i in 0 until size) {
        val command = readLine()!!.split(" ")
        when(command[0]) {
            "push" -> stack.push(command[1].toInt())
            "top" -> println(stack.top())
            "size" -> println(stack.size())
            "empty" -> println(stack.empty())
            "pop" -> println(stack.pop())
        }
    }
}